### ISSUE TITLE
大佬！微信捐赠快速打开微信付款界面是怎么实现的？源码没看懂第三行

### ISSUE THREAD

**Issue by xiaofei-dev** [veryyoung/QQLuckyMoney#70]
```  
     Intent localIntent = new Intent();
     localIntent.setClassName("com.tencent.mm", "com.tencent.mm.ui.LauncherUI");
     localIntent.putExtra("donate", true);
     SettingsActivity.SettingsFragment.this.startActivity(localIntent);
```

如代码所示，并没有看到有什么个人信息设置，为什么跳转到微信后直接就打开给你付款的界面了？？
求解惑！

**Comment by veryyoung** [comment]
@xiaofei-dev 这里相当于打开微信，传了一个 ```donate```的参数

**Comment by veryyoung** [comment]
`   if (loadPackageParam.packageName.equals(WECHAT_PACKAGE_NAME)) {
            findAndHookMethod("com.tencent.mm.ui.LauncherUI", loadPackageParam.classLoader, "onCreate", Bundle.class, new XC_MethodHook() {
                @Override
                protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                    Activity activity = (Activity) param.thisObject;
                    if (activity != null) {
                        Intent intent = activity.getIntent();
                        if (intent != null) {
                            String className = intent.getComponent().getClassName();
                            if (!TextUtils.isEmpty(className) && className.equals("com.tencent.mm.ui.LauncherUI") && intent.hasExtra("donate")) {
                                Intent donateIntent = new Intent();
                                donateIntent.setClassName(activity, "com.tencent.mm.plugin.remittance.ui.RemittanceUI");
                                donateIntent.putExtra("scene", 1);
                                donateIntent.putExtra("pay_scene", 32);
                                donateIntent.putExtra("fee", 10.0d);
                                donateIntent.putExtra("pay_channel", 13);
                                donateIntent.putExtra("receiver_name", "yang_xiongwei");
                                donateIntent.removeExtra("donate");
                                activity.startActivity(donateIntent);
                                activity.finish();
                            }
                        }
                    }
                }
            });
        }
`

这里是真正启动捐赠的地方

**Comment by xiaofei-dev** [comment]
谢前辈解答[抱拳]。

**Comment by uchen1225** [comment]
qq 6.71 is coming.
i beg you update.thanks in advance.

**Comment by veryyoung** [comment]
@uchen1225  done

**Comment by uchen1225** [comment]
thanks a lot

2017年3月30日 下午3:32，"veryyoung" <notifications@github.com>写道：

**Comment by uchen1225** [comment]
thanks a lot

2017年3月30日 下午3:32，"veryyoung" <notifications@github.com>写道：

Closed #70 <https://github.com/veryyoung/QQLuckyMoney/issues/70>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/veryyoung/QQLuckyMoney/issues/70#event-1021787353>, or mute
the thread
